Objects/rangeobject.c

```diff
@@ -576,13 +576,19 @@ range_index(rangeobject *r, PyObject *ob)
         return NULL;
 
     if (contains) {
-        PyObject *idx, *tmp = PyNumber_Subtract(ob, r->start);
-        if (tmp == NULL)
+        PyObject *idx = PyNumber_Subtract(ob, r->start);
+        if (idx == NULL) {
             return NULL;
+        }
+
+        if (r->step == _PyLong_One) {
```

Comment:
Sadly, I made this check a little bit less efficient in commit c9bc290dd6e3994a4ead2a224178bcba86f0c0e4 to fix bpo-42161.